org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AbstractRunnableDeviceDelegate.java

```diff
@@ -0,0 +1,183 @@
+ * <P>
+ * Since this defines all functions through inheritance rather than through annotations, and it doesn't depend on
+ * interface default methods, it can be used as a base class for delegates defined in Jython.
+ * <P>
+ * Note that an object of this class must be initialised with a reference to the object it is going to provide
+ * delegated functionality to and setDelegate() on that object should then be immediately called with this one.
+ * <p>
+ * Example of use:
+ * </P>
+ * <pre>{@code
+areaDetectorRunnableDeviceProxyFinder = finder.find("areaDetectorRunnableDeviceProxyFinder")
+areaDetectorRunnableDeviceProxy = areaDetectorRunnableDeviceProxyFinder.getRunnableDevice()
+
+from jythonAreaDetectorRunnableDeviceDelegate import JythonAreaDetectorRunnableDeviceDelegate
+jythonAreaDetectorRunnableDeviceDelegate = JythonAreaDetectorRunnableDeviceDelegate(areaDetectorRunnableDeviceProxy)
+areaDetectorRunnableDeviceProxy.setDelegate(jythonAreaDetectorRunnableDeviceDelegate)
+areaDetectorRunnableDeviceProxy.register()
+ * }</pre>
+ */
+public abstract class AbstractRunnableDeviceDelegate {
+
+	private static final Logger logger = LoggerFactory.getLogger(AbstractRunnableDeviceDelegate.class);
+
+	RunnableDeviceProxy runnableDeviceProxy;
+
+	public AbstractRunnableDeviceDelegate(RunnableDeviceProxy runnableDeviceProxy) {
+		this.runnableDeviceProxy = runnableDeviceProxy;
+	}
+
+	public RunnableDeviceProxy getRunnableDeviceProxy() {
+		return runnableDeviceProxy;
+	}
+
+	// Delegated AbstractRunnableDevice<AreaDetectorRunnableDeviceModel> methods
+
+	@SuppressWarnings("unused")
+	public void configure(Object model) throws ScanningException {
+		logger.trace("configure({}) on {}", model, runnableDeviceProxy.getName());
+	}
+
+	// Delegated interface IRunnableDevice<AreaDetectorRunnableDeviceModel> methods
+
+	@SuppressWarnings("unused")
+	public void run(IPosition position) throws ScanningException, InterruptedException {
+		logger.trace("run({}) on {}", position, runnableDeviceProxy.getName());
+	}
+
+	// Delegated interface IWritableDetector<AreaDetectorRunnableDeviceModel> methods
+
+	@SuppressWarnings("unused")
+	public boolean write(IPosition position) throws ScanningException {
+		logger.trace("write({}) on {}", position, runnableDeviceProxy.getName());
+		return false;
+	}
+
+	// Delegated interface INexusDevice<NXdetector> methods
+
+	@SuppressWarnings("unused")
+	public NexusObjectProvider<NXdetector> getNexusProvider(NexusScanInfo info) throws NexusException {
+		logger.trace("getNexusProvider({}) on {}", info, runnableDeviceProxy.getName());
+		return null;
+	}
+
+
+	protected NexusObjectWrapper<NXdetector> getNexusObjectWrapper(String name, NXdetector nexusBaseClass) {
+		return new NexusObjectWrapper<NXdetector>(name, nexusBaseClass);
+	}
+
+	// Delegated annotated methods
+
+	@SuppressWarnings("unused")
+	public void preConfigure(Object scanModel, ScanBean scanBean, IPublisher<?> publisher) throws ScanningException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.eclipse.scanning.api.scan.ScanningException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)